### PR TITLE
Fix dev environment

### PR DESCRIPTION
**Pull request by immora**
To fix a bug in **client-kundenportal**, I used this repo to set up the backend part. 

This was unfortunately very hard, but finally it worked and this PR is a record of the changes I needed to make.

The biggest problems was that config files with ".dev." were not used, but still some parts were configured to use them...

### CODE REVIEW

File: config/server/openolitor-server.dev.conf [hunk 1]
```diff
@@ -303,7 +302,7 @@ openolitor {
       password = "1234"
       security= "STARTTLS"
       number-of-retries = 3
-      send-email = false
+      send-email = true
```
Comment (immora) [inline config/server/openolitor-server.dev.conf:305]: Without this I couldn't create a user.

File: docker-compose.yml [hunk 2]
```diff
@@ -20,8 +20,8 @@ services:
     volumes:
       - ./data/s3:/data
     environment:
-      - MINIO_ACCESS_KEY=miniouserallcsa
-      - MINIO_SECRET_KEY=thedefaultminiosecretneedstobechanged
+      - MINIO_ACCESS_KEY=openolitor
+      - MINIO_SECRET_KEY=uGuRR53g4uR6c8R4yCxfNwFF
```
Comment (immora) [inline docker-compose.yml:24]: these credentials were the actual ones expected

**Comment by mcmpp** [comment]
Hi @immora ,

that's so great that you went so far!! I am glad to see you onboard with such a motivation. You are absolutely right, there were some mismatches among the documentation and the project that were not helping on getting everything working easily. I see you hacked quite a lot to get it running :) 

I see that you made changes into the dev files as well as the default ones. The default ones are supposed to work from localhost (see readme section "Run on localhost"). However that was not the case. I made some changes and create a pull request with a fixed version https://github.com/OpenOlitor/openolitor-docker-compose/pull/3. Please, have a look

On the other side, there is an automatic generation of the config files for different environments. The dev files are an example of this case (see readme section "Generate Configuration"). In that case the dev configuration files are overwritten with the information we provide on the dev.json file while running the generator.py. 

What is your case?? I am assuming you are trying to run it in localhost but I might be wrong... 
Please, any other suggestion is more than welcome. And, please, if you have any difficulty just contact us. We can help and learn together.

Have a great day and I hope I am reading from you soon

**Comment by mcmpp** [comment]
Great, thanks for your work!! 

Are you part of one initiative using openolitor?? 

Mikel

**Comment by immora** [comment]
Hah, actuall yes - I'm a subscriber of Gemüse Abo of Brunner Eichhof here in Bern ;)

**Comment by mcmpp** [comment]
Hi @immora,

I have merged the pull request and verified that the default version works. 

Thanks so much for your feedback

**Comment by immora** [comment]
Great, @mcmpp! I'm very glad. Then I guess this PR can be closed :)

**Pull request closed without merging**